pkg/controller/master/backup/backup_target.go

```diff
@@ -79,32 +80,79 @@ func (h *TargetHandler) OnBackupTargetChange(key string, setting *harvesterv1.Se
 		return setting, err
 	}
 
-	// Since we remove S3 access key id and secret access key after S3 backup target has been verified,
-	// we stop the controller to reconcile it again.
-	if target.Type == settings.S3BackupType && (target.SecretAccessKey == "" || target.AccessKeyID == "") {
-		return nil, nil
-	}
+	logrus.Debugf("backup target change:%s:%s", target.Type, target.Endpoint)
 
-	if err = h.updateLonghornTarget(target); err != nil {
-		return nil, err
-	}
+	switch target.Type {
+	case settings.S3BackupType:
+		// Since S3 access key id and secret access key are stripped after S3 backup target has been verified
+		// in reUpdateBackupTargetSettingSecret
+		// stop the controller to reconcile it again.
+		if target.SecretAccessKey == "" && target.AccessKeyID == "" {
+			return nil, nil
+		}
+
+		if err = h.updateLonghornTarget(target); err != nil {
+			return nil, err
+		}
 
-	if target.Type == settings.S3BackupType {
 		if err = h.updateBackupTargetSecret(target); err != nil {
 			return nil, err
 		}
-	}
 
-	return h.updateBackupTargetSetting(setting.DeepCopy(), target, err)
+		return h.reUpdateBackupTargetSettingSecret(setting, target, err)
+
+	case settings.NFSBackupType:
+		if err = h.updateLonghornTarget(target); err != nil {
+			return nil, err
+		}
+
+		// delete the may existing previous secret of S3
+		if err = h.deleteBackupTargetSecret(target); err != nil {
+			return nil, err
+		}
+
+		// same operation as S3 does, but it looks to be not as expected
+		// the framework will set the True/False according to the final return of here
+		harvesterv1.SettingConfigured.SetError(setting, "", err)
```

Comment:
remove it since it will be updated by https://github.com/harvester/harvester/blob/master/pkg/controller/master/setting/handler.go#L83

Reply:
> remove it since it will be updated by https://github.com/harvester/harvester/blob/master/pkg/controller/master/setting/handler.go#L83

It is added to keep logic same as before, because in s3 path, it called this API. 
I also wondered, this explicit setting has no effect.

Reply:
updated